webui/react/src/components/kit/CodeEditor.tsx

```diff
@@ -30,18 +27,27 @@ import { ErrorType } from 'shared/utils/error';
 import { AnyMouseEvent } from 'shared/utils/routes';
 import handleError from 'utils/error';
 
-const JupyterRenderer = lazy(() => import('./IpynbRenderer'));
+const JupyterRenderer = lazy(() => import('./CodeEditor/IpynbRenderer'));
 
 const { DirectoryTree } = Tree;
 
-import css from './CodeViewer.module.scss';
+import css from './CodeEditor/CodeEditor.module.scss';
 
-import './index.scss';
+import './CodeEditor/index.scss';
+
+type FileInfo = {
+  content: string;
+  name: string;
+};
 
 export type Props = {
-  experimentId: number;
+  editable?: boolean;
```

Comment:
quibble -- given that the name of the component is `CodeEditor`, maybe we should flip it so it's editable by default and in the case of wanting the code editor to not be editable, we can use `readonly` instead?

Reply:
✅